templates/include/telemetry/environment.1.schema.json

```diff
@@ -820,6 +820,12 @@
                   },
                   "GPUActive": {
                     "type": "boolean"
+                  },
+                  "adapterDriver": {
+                    "type": [
+                      "string",
+                      "null"
+                    ]
```

Comment:
Do you have an example query showing instances of `adapterDriver`? I tried the `main` and `event` ping (https://sql.telemetry.mozilla.org/queries/69068/source). Additionally, I don't see the field listed in the source docs: https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/data/environment.html#environment

Reply:
This would be one example query: 
https://console.cloud.google.com/bigquery?sq=1018637472215:73ba37ca3a3445519443811af49da76d

Right, it doesn't seem to be documented anywhere. But it gets send when looking into the code: https://searchfox.org/mozilla-central/source/toolkit/components/telemetry/app/TelemetryEnvironment.jsm#509

Reply:
This may be a typo then, `adapterDriver` should be renamed into `driverVendor`, which is a documented field.

Reply:
Mh, it seems like both fields are getting used. But right, if `driverVendor` is set then `adapterDriver` is null and vice versa.

Reply:
I'm a bit unclear on the usage of `adapterDr. In the cited `TelemetryEnvironment::getGfxAdapter` method, it looks like the function is returning a single adapter element. Internally, there is a property `adapterDriver` appended with a suffix. It seems like this is translated into the fields observed in in this section of the schema by removing the adapter prefix (e.g. `adapterDriver` seems to map to `environment.system.gfx.adapters.[].driver`). I [modified the query to search for `adapterDriver`](
https://console.cloud.google.com/bigquery?sq=1018637472215:73ba37ca3a3445519443811af49da76d)
in `additional_properties`, which returns no rows. Is there an `adapterDriver` field that you've found in one of the other pings? 

The query example you provided extracts `$.environment.system.gfx.adapters[0].driverVendor` from `additional_properties`, which does seems to be missing. 

It may be useful for review to update a validation document with a value observed from `additional_properties` for the missing field(s).

Reply:
I don't see a `adapterDriver` field in any other ping. I would also think that `adapterDriver` maps to `environment.system.gfx.adapters.[].driver`.

I'll add this case to one of the validation documents 👍